packages/binding-coap/src/coap-server.ts

```diff
@@ -524,9 +525,9 @@ export default class CoapServer implements ProtocolServer {
         const listener = this.createContentListener(req, res, this.PROPERTY_DIR, affordanceKey);
 
         try {
-            await thing.handleObserveProperty(affordanceKey, listener, null);
```

Comment:
@relu91 Hmm, so far the binding implementation always passed `null` to the `handleObserveProperty` and `handleUnobservProperty` methods. Not sure why that is the case. I will have another look at this in the binding itself and keep the API in core untouched then :)

Reply:
I think the latest commits should now have provided the correct fix for the issue :) I think that this should also make the behavior of the server more predictable, as the form index was apparently not taken into account before when a property was observed.